docs_src/async_sql_databases/tutorial001.py

```diff
@@ -58,8 +58,43 @@ async def read_notes():
     return await database.fetch_all(query)
 
 
+@app.get("/notes/{note_id}/", response_model=Note)
+async def read_one_note(note_id: int):
+    query = notes.select().where(notes.c.id == note_id)
+    note = await database.fetch_one(query)
+    if note is None:
+        raise HTTPException(status_code=404, detail="Note not found")
+    return note
+
+
 @app.post("/notes/", response_model=Note)
 async def create_note(note: NoteIn):
-    query = notes.insert().values(text=note.text, completed=note.completed)
+    query = notes.insert().values(**note.dict())
     last_record_id = await database.execute(query)
     return {**note.dict(), "id": last_record_id}
+
+
+@app.put("/notes/{note_id}/", response_model=Note)
+async def update_note(note_id: int, note: NoteIn):
+    query = notes.select().where(notes.c.id == note_id)
+    note_ = await database.fetch_one(query)
+    if note_ is None:
+        raise HTTPException(status_code=404, detail="Note not found")
+    query = (
+        notes.update()
+        .values(**note.dict())
+        .where(notes.c.id == note_id)
+    )
+    await database.execute(query)
+    query = notes.select().where(notes.c.id == note_id)
```

Comment:
I would probably just return a `200` here. Making another query to get the note you just updated seems unnecessary to me.

Reply:
@falkben, yes, I have done that in the latest commit.